conftest.py

```diff
@@ -12,40 +12,18 @@
 """
 
 import os
-import os.path as osp
-import sys
-
-import pylsp_utils
 
 # ---- To activate/deactivate certain things for pytest's only
 # NOTE: Please leave this before any other import here!!
 os.environ['SPYDER_PYTEST'] = 'True'
 
-
-# ---- Detect if we're running in CI
-# Note: Don't import from spyder to keep this file free from local
-# imports.
-running_in_ci = bool(os.environ.get('CI'))
-
-
-# ---- Handle subrepos
-# NOTE: Please don't move this from here!
+# ---- Update PyLSP
 # When running in CI, subrepos are installed in the env.
-if not running_in_ci:
-    HERE = osp.dirname(osp.abspath(__file__))
-    DEPS_PATH = osp.join(HERE, 'external-deps')
-    i = 0
-    for path in os.listdir(DEPS_PATH):
-        external_dep_path = osp.join(DEPS_PATH, path)
-        sys.path.insert(i, external_dep_path)
-        i += 1
-
-    # Remove previous local PyLSP installation.
-    pylsp_utils.remove_installation()
-
-    # Install PyLS locally.
-    pylsp_utils.install()
-
+if not os.environ.get('CI'):
+    from install_subrepos import REPOS, install_repo
+    if REPOS['python-lsp-server']['editable']:
+        # Installed in editable mode -> reinstall
+        install_repo('python-lsp-server', editable=True)
```

Comment:
I agree that this is not desirable. My understanding is that the python-lsp-server has an executable which may be called, so if it is installed in editable mode, the executable will need to be rebuilt. Perhaps there's a way to do this conditionally if the repo has changed? Or maybe this is a non-issue? in which case we could remove it entirely from `bootstrap.py` and `conftest.py`. Note, that under the current bootstrap, the executable is _not_ rebuilt after install; I don't know if this has caused any issues or not when checking out different commits that span different python-lsp-server changes.

Reply:
Unless there is some very esoteric somewhere, after confirming in `python-lsp-server`'s packaging config, there aren't any compiled files that would need to be rebuilt (other than Python's own bytecode, which is handled at runtime). The only thing I believe you could be referring to are the entry points, but changing the name of the main CLI entrypoint (`pylsp`) would be a very rare/unlikely change and it just points to `__main__`, which would never change; there are the various plugin entrypoint names and module paths that could theoretically change, but that would also be a breaking change and thus not likely to be too common, and be much less of an issue (as only that one plugin would be affected).

Therefore, given it apparently hasn't been an issue thus far, I don't think it justifies making the Spyder startup time go from 0.9 s to 43 s every time, when the `install_dev_repos.py` script can just be re-run in case there are major changes.

Reply:
Upon further investigation, I concur with your conclusion: we don't need this in `bootstrap.py` or `conftest.py` and I will remove it.

When I first started updating bootstrap, I noticed that _only_ `python-lsp-server` was being reinstalled on bootstrap and in the test configuration. I figured this was due to the "executable" built at `<env>/bin/pyslp`. I didn't actually notice that it is not a binary file, but a script with executable permissions. So, unless the content of that script changes, there should not be any need to reinstall `python-lsp-server`. There must have been some reason, now defunct, for reinstalling this package but not any of the others. 🤷

Reply:
Yeah, that's just the  `console_script` wrapper pip generates per the entrypiont spec, which shouldn't change unless either the `console_script` entrypoint name/value in the `setup.cfg` changes (which it shouldn't, ever) or pip updates how it generates them, which would only be relevant reinstalling with a new version of pip anyway.

Perhaps the plugin entrypoints? Which should be pretty stable now...